server/src/tests/socketio.test.ts

```diff
@@ -1,146 +1,188 @@
 // Testing for socket.io endpoints
 
-import { io as io } from 'socket.io-client'
-import { v4 as uuidv4 } from 'uuid';
-import { Message } from '../types/Message';
+import { createServer } from "node:http";
+import { io as ioc } from "socket.io-client";
+import { Server } from "socket.io";
+import { v4 as uuidv4 } from "uuid";
 
-const socket_test_client_port = process.env.socket_test_client_port;
-console.log("Socket clients are listening on port", socket_test_client_port)
-
-const SECONDS_TIMEOUT = 10;
 const SECONDS_MULTIPLIER = 1000;
-const NUM_CLIENTS = 3; // Adjust the number of clients as needed. Do not go over 300 to prevent being blocked by Firebase.
-const exampleMsg: Message = {
-    uid: uuidv4(), // random Ids; not a real UID
-    msgId: uuidv4(),
-    msgContent: "MESSAGE CONTENT",
-    timeSent: 9999,
-    location: {
-        lat: 10,
-        lon: 10
-        // Geohash will be calculated by the server since it is not included with the message.
+jest.setTimeout(60 * SECONDS_MULTIPLIER);
+
+describe("socket-load-tests", () => {
+  let clientSockets = [];
+  let httpServer;
+  let httpServerAddr;
+  let ioServer;
+  const numClients = 100; // Adjust the number of clients as needed. Do not go over 300 to prevent being blocked by Firebase.
```

Comment:
Just as a heads up, if we are using this test on every PR, with one firebase account, its going to rack up a lot of requests, so I feel like this number should be lowered. So I feel about 50 or less connections is good enough for load testing, I really don't believe any more than that will be connecting and sending messages at once.

Though I do understand that having 100 or so is a good attempt at seeing what the server can handle.

Reply:
Will lower :+1: